app/src/main/java/swati4star/createpdf/activity/WelcomeActivity.java

```diff
@@ -0,0 +1,155 @@
+package swati4star.createpdf.activity;
+
+import android.content.Context;
+import android.content.Intent;
+import android.os.Bundle;
+import android.support.v4.view.PagerAdapter;
+import android.support.v4.view.ViewPager;
+import android.support.v7.app.AppCompatActivity;
+import android.text.Html;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.Button;
+import android.widget.LinearLayout;
+import android.widget.TextView;
+import android.widget.Toast;
+
+import swati4star.createpdf.R;
+
+public class WelcomeActivity extends AppCompatActivity {
+    private ViewPager mViewPager;
+    private MyViewPagerAdapter mViewPagerAdapter;
+    private LinearLayout mDotsLayout;
+    private TextView[] mDots;
+    private int[] mLayouts;
+    private final int mTimeDelay = 2000;
+    private long mBackPressed;
+    private Button mBtnSkip;
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+
+
+        setContentView(R.layout.activity_welcome);
+
+        mViewPager = (ViewPager) findViewById(R.id.view_pager);
+        mDotsLayout = (LinearLayout) findViewById(R.id.layoutDots);
+        mBtnSkip = findViewById(R.id.btn_skip);
+
+
+
+        // layouts of all welcome sliders
+        // add few more layouts if you want
+        mLayouts = new int[]{
+                R.layout.fragment_intro_1,
+                R.layout.fragment_intro_3,
+                R.layout.fragment_intro_4,
+                R.layout.fragment_intro_2 };
+
+        // adding bottom dots
+        addBottomDots(0);
+
+        mViewPagerAdapter = new MyViewPagerAdapter();
+        mViewPager.setAdapter(mViewPagerAdapter);
+        mViewPager.addOnPageChangeListener(viewPagerPageChangeListener);
+
+        mBtnSkip.setOnClickListener(view -> {
+            Intent intent = new Intent(getApplicationContext(), MainActivity.class);
+            startActivity(intent);
+        });
+    }
+
+    private void addBottomDots(int currentPage) {
```

Comment:
add comments to the newly added function

Reply:
alright noted